### DISCUSSION TITLE
Meta Data in Child Flows

### DISCUSSION THREAD

**Discussion by danielband** [directus/directus#18743] [Feature Requests]
### Summary

I started to replace most of my action hooks in favor of Flows + custom operations. What I am missing though, is a way to pass meta data down the flow chain. E.g. I created my own little database backed event bus to react on events that happen during the execution of the flow. To configure the event data, each parent flow execution creates a uuid and a context string right after the initial trigger; so I know what happens during the flow. I pass this data to every operation. When something important happens the operation creates a new item in the event table. In another flow a hook is listening on configurable events and sends out notifications.

Right now, I pass this event configuration to the payload of the child flow. In case, the payload is an array, I have to map payload and append the config to every item, before I call the child flow.

Would it be possible to add a meta data field in the child flow, so common data can be passed without the need of a script/transform operation?

### Motivation

Make flows leaner and thus easier to read. Also it would save time.

### Detailed Design

Add a meta data field to the flow operation. The data should be passed to every flow execution. Maybe it is enough to merge the meta data with the payload in the operation handler.

### Must Have:
- Flow operation needs a new field 'meta data'
- The data should get passed to children

### Drawbacks

I might be able to implement this myself. Obviously only, if this feature is something you guys want to integrate.

### Alternatives

Right now, I pass this event configuration to the payload of the child flow. In case, the payload is an array, I have to map payload and append the config to every item, before I call the child flow.

I tried to put the meta data in accountability, but that fails because of type warnings. 

### Adoption Strategy

This should not cause any breaking change imo.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
